src/components/menus/ExtensionMenu.svelte

```diff
@@ -0,0 +1,68 @@
+<svelte:options accessors={true} immutable={true} />
+
+<script lang="ts">
+  import SettingsIcon from '@nasa-jpl/stellar/icons/settings.svg?component';
+  import { createEventDispatcher } from 'svelte';
+  import type { User } from '../../types/app';
+  import type { Extension } from '../../types/extension';
+  import { permissionHandler } from '../../utilities/permissionHandler';
+  import PlanNavButton from '../plan/PlanNavButton.svelte';
+  import MenuItem from './MenuItem.svelte';
+
+  export let extensions: Extension[];
+  export let user: User | null;
+
+  const dispatch = createEventDispatcher();
+
+  function callExtension(extension: Extension) {
+    dispatch('callExtension', extension);
+  }
+
+  function hasExtensionPermission(extension: Extension): boolean {
+    for (const extensionRole of extension.extension_roles) {
+      if (user?.activeRole === extensionRole.role) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+</script>
+
+{#if extensions.length > 0}
+  <div class="view-menu st-typeography-medium">
+    <PlanNavButton title="Extensions" menuTitle="">
+      <SettingsIcon />
+      <div class="view-menu st-typography-medium" slot="menu">
+        {#each extensions as extension}
+          <MenuItem
+            on:click={() => callExtension(extension)}
+            use={[
+              [
+                permissionHandler,
+                {
+                  hasPermission: hasExtensionPermission(extension),
+                  permissionError: 'You do not have permission to call this extension',
+                },
+              ],
+            ]}
+          >
+            {extension.label}
+          </MenuItem>
+        {/each}
+      </div>
+    </PlanNavButton>
+  </div>
+
+  <style>
+    .view-menu {
+      --aerie-menu-item-template-columns: auto;
+      align-items: center;
+      cursor: pointer;
+      display: grid;
+      height: inherit;
+      justify-content: center;
+      position: relative;
+    }
```

Comment:
May want to add this CSS to hide the menu body (also done in ViewMenu)
```CSS
.view-menu :global(.header),
  .view-menu :global(.nav-button .menu-body) {
    display: none;
  }
```

Reply:
Though something funny is happening with the CSS/classes here and in view menu...

Reply:
Aha, need to move the `<style>` block outside of the `{#if` block. Also I'd recommend not re-using the `view-menu` class twice (know ViewMenu does it, we should fix that). So perhaps `extensions-menu` and `extensions-menu--menu` or something.

Reply:
Yeah I didn't realize I put it in there, good catch! I'll rename the class.